### PR TITLE
fix(backend): known_members don't filter deleted users and workspaces

### PR DESCRIPTION
**Pull request by Millefeuille42**
#6286
#6287
#6289

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [ ] Manual, quality tests have been done

### CODE REVIEW

File: backend/tracim_backend/lib/core/user.py [hunk 1]
```diff
@@ -267,7 +268,11 @@ def get_users_in_common_with_user_workspace(self, user_id: int) -> List[Tuple[An
         return (
             self._session.query(s2.user_id, User.username, User.display_name, s2.workspace_id)
             .join(s1, s1.workspace_id == s2.workspace_id)
```
Comment (floreal15) [inline backend/tracim_backend/lib/core/user.py:270]: using .filter(s1.user_id == user_id) here should greatly reduce the request time

Comment (Millefeuille42) [reply backend/tracim_backend/lib/core/user.py]: Done, it works !

**Comment by Millefeuille42** [comment]
It's a bother that we have to do an additional join on workspaces, it slows down the request a lot unfortunately and it has low to no impact on the overall behaviour.

However, I fear that this will turn into a "future us" issue someday.

File: backend/tracim_backend/lib/core/user.py [hunk 2]
```diff
@@ -395,7 +399,7 @@ def get_known_users(
                 User.email.ilike(f"%{acp}%"),
                 User.username.ilike(f"%{acp}%"),
             )
-        )
+        ).filter(and_(User.is_active, not_(User.is_deleted)))
```
Comment (buxx) [inline backend/tracim_backend/lib/core/user.py:402]: Je comprends que ce filtre : 

* n'inclus que les utilisateurs actifs
* exclut les utilisateurs supprimés

Je pense que nous ne devrions pas exclure les utilisateurs inactifs : Lorsque l'on ajoute des utilisateurs Tracim, on souhaite probablement pouvoir les ajouter à des espaces avant qu'ils n'aient activé leur compte.

File: backend/tracim_backend/lib/core/user.py [hunk 3]
```diff
@@ -266,9 +267,12 @@ def get_users_in_common_with_user_workspace(self, user_id: int) -> List[Tuple[An
 
         return (
             self._session.query(s2.user_id, User.username, User.display_name, s2.workspace_id)
-            .join(s1, s1.workspace_id == s2.workspace_id)
-            .join(User, User.user_id == s2.user_id)
-            .filter(s1.user_id == user_id)
+            .join(s1, and_(s1.workspace_id == s2.workspace_id, s1.user_id == user_id))
+            .join(User, and_(User.user_id == s2.user_id, User.is_active, not_(User.is_deleted)))
```
Comment (buxx) [inline backend/tracim_backend/lib/core/user.py:271]: Idem que le commentaire précédent : est-ce ok d'exclure les utilisateurs dont le compte n'est pas encore actif.

File: backend/tracim_backend/lib/core/user.py (hunk 2 above)
Comment (Millefeuille42) [reply backend/tracim_backend/lib/core/user.py:402]: Ok !

File: backend/tracim_backend/lib/core/user.py (hunk 3 above)
Comment (Millefeuille42) [reply backend/tracim_backend/lib/core/user.py:271]: Ça pourrait poser des soucis de RGPD de donner l'accès à des données. Pour le moment on va le laisser tel quel mais adresser le problème plus tard.

Comment (buxx) [reply backend/tracim_backend/lib/core/user.py:271]: Oui c'est un autre sujet en effet.

**Review by buxx — APPROVED** [review]

**Comment by buxx** [comment]
@Millefeuille42 Approuvé (avec le retrait des filtres sur is_active)

**Comment by buxx** [comment]
@Millefeuille42 N'oublie pas `User.is_active` dans `.join(User, and_(User.user_id == s2.user_id, User.is_active, not_(User.is_deleted)))`  :D

**Comment by lebouquetin** [comment]
@Millefeuille42 working on this code is the opportunity to fix the typing (`Any` is useless for typing)

**Comment by Millefeuille42** [comment]
will change this and also the Any typing

**Comment by Millefeuille42** [comment]
Tests are ok since only unstable e2e don't pass

**Pull request merged**